taskvine/src/manager/vine_manager.c

```diff
@@ -2873,27 +2876,40 @@ assignment and the new task state.
 
 static vine_result_code_t commit_task_to_worker(struct vine_manager *q, struct vine_worker_info *w, struct vine_task *t)
 {
-	/* Kill empty libraries to reclaim resources. Match the assumption of
-	 * @vine.schedule.c:check_worker_have_enough_resources() */
+	vine_result_code_t result = VINE_SUCCESS;
+
+	/* Kill unused libraries on this worker to reclaim resources. */
+	/* Matches assumption in vine_schedule.c:check_available_resources() */
 	kill_empty_libraries_on_worker(q, w, t);
```

Comment:
I think this kill should be done unconditionally only for tasks that do not require a library. Then, for tasks that require a library it should be done only if the check at line 2889 fails. Otherwise for workers running one-slot libraries, wouldn't the libraries get kill and re-send every time?

Reply:
The expression is a bit twisted, but I think the effect of line 2865 is to *not* kill the library if the incoming function needs it.  Am I missing something?

The logic is a bit opaque, perhaps the expression should be rewritten to make it more clear.

Reply:
You are right, now I see that kill_empty_libraries does the right thing.